sdk/eventhub/event-hubs/src/partitionProcessor.ts

```diff
@@ -12,54 +53,170 @@ import { ReceivedEventData } from "./eventData";
  * - Optionally override the `close()` method to implement any tear down or clean up tasks you would want to carry out.
  */
 export class PartitionProcessor {
+  private _partitionManager: PartitionManager | undefined;
+  private _consumerGroupName: string | undefined;
+  private _eventHubName: string | undefined;
+  private _eventProcessorId: string | undefined;
+  private _partitionId: string | undefined;
+  private _eTag: string = "";
+  /**
+   * @property The consumer group name
+   * @readonly
+   */
+  public get consumerGroupName() {
+    return this._consumerGroupName!;
+  }
+
+  /**
+   * @property The event hub name
+   * @readonly
+   */
+  public get eventHubName() {
+    return this._eventHubName!;
+  }
+
+  /**
+   * @property The identifier of the Event Hub partition
+   * @readonly
+   */
+  public get partitionId() {
+    return this._partitionId!;
+  }
+  /**
+   * @property The unique identifier of the event processor
+   * @readonly
+   */
+  public get eventProcessorId() {
+    return this._eventProcessorId!;
+  }
+
+  /**
+   * @property An instance of `PartitionManager`
+   * @readonly
+   */
+  public get partitionManager() {
+    return this._partitionManager!;
+  }
+
+  /**
+   * @property The consumer group name
+   */
+  public set consumerGroupName(consumerGroupName: string) {
```

Comment:
Setters should not be public.

Reply:
In TypeScript getters and setters have to have the same visibility. So either we can set both public or both private.

Reply:
User cannot set anything because I added a check in all the setters, if the private field is set then do nothing.